dashboard/app/api.go

```diff
@@ -751,18 +751,21 @@ func reportCrash(c context.Context, build *Build, req *dashapi.Crash) (*Bug, err
 	} else if len(req.ReproSyz) != 0 {
 		reproLevel = ReproLevelSyz
 	}
-	newSubsystems := []string{}
+	newSubsystems := []*subsystem.Subsystem{}
 	save := reproLevel != ReproLevelNone ||
 		bug.NumCrashes < int64(maxCrashes()) ||
 		now.Sub(bug.LastSavedCrash) > time.Hour ||
 		bug.NumCrashes%20 == 0 ||
 		!stringInList(bug.MergedTitles, req.Title)
 	if save {
-		newSubsystems = detectCrashSubsystems(c, req, build)
-		log.Infof(c, "determined subsystems: %q", newSubsystems)
 		if err := saveCrash(c, ns, req, bug, bugKey, build, assets); err != nil {
 			return nil, err
 		}
+		newSubsystems, err = inferSubsystems(c, bug, bugKey)
```

Comment:
It does not look right to load a bunch of previous crashes every time we save a new one. We don't need to update subsystems every second. Can we do this somewhere else?

Reply:
In some cases we definitely want to update the subsystems ASAP, e.g. if we receive an interesting reproducer that'll help us determine the exact guilty filesystem implementation. The decision must be made before the bug is reported.

Maybe recalculate them in `reportCrash` for the first crash and every saved reproducer? This should be much less often than every saved crash.